### ISSUE TITLE
DOC: no docs for ExcelWriter `engine_kwargs` #42292 #42214

### ISSUE THREAD

**Issue by Joeperdefloep** [pandas-dev/pandas#43440]
### 

- [X] I have checked that the issue still exists on the latest versions of the docs on `master` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/dev/reference/api/pandas.ExcelWriter.html?highlight=excelwriter#pandas.ExcelWriter

### Documentation problem

There is no use-case specified for `engine_kwargs`, as a result of which I was confused and made #42214. There, my test was overwritten by #42292 that uses the 'correct' use of `engine_kwargs`. How I did it first (incorrect):
```
import pandas as pd
with pd.ExcelWriter("file.xlsx",engine="xlsxwriter",engine_kwargs={"strings_to_formulas":False}) as writer:
    pd.DataFrame(["=1+1"]).to_excel(writer)
```
where it should have been:
```
import pandas as pd
with pd.ExcelWriter("file.xlsx",engine="xlsxwriter",engine_kwargs={"options":{"strings_to_formulas":False}}) as writer:
    pd.DataFrame(["=1+1"]).to_excel(writer)
```
However, since only `xlsxwriter` actually passes the `engine_kwargs` through to the `Workbook` class, this is unintuitive without documentation.

### Suggested fix for documentation

I will submit a PR

**Comment by acse-srm3018** [comment]
Hi. This is my first contribution to an open source project. Can I take this issue?

**Comment by Joeperdefloep** [comment]
It's already solved by #43445. Thank you for contributing! This is really important. Normally issues you can take up have a 'contributions welcome' tag you can search for. 'Good first issue' means it is a good issue that you do not need very in-depth knowledge of Pandas of, so that is also a good place to start :)
And yes, then you comment

> Take!

On an issue to notify people that you are working on that and they do not get double-solved. Here I said

> I will submit a PR

Which is the same. Good luck and thanks!

Also please read the contributing guide, there this is explained.
